src/odemis/driver/actuator.py

```diff
@@ -2896,17 +2896,28 @@ class DualChannelPositionSensor(model.HwComponent):
     .reference: call to sensor adjustment routine
     """
 
-    def __init__(self, name, role, dependencies, channels, distance, **kwargs):
+    def __init__(self, name, role, dependencies, channels, distance, ref_on_init=False, **kwargs):
         """
         dependencies: (dict str --> Component) dict with "sensor" key containing three-channel sensor component.
-            The three-channel sensor is required to have a .channels attribute, a .position VA and
+            The three-channel sensor is required to have a .axes attribute, a .position VA and
             .reference and .stop functions.
+            Optionally, a "stage" dependency can be passed. All its axes will be referenced when calling reference.
         channels: (dict str --> str, [str], or [str, str]) mapping of output channels to sensor channels, one output
             channel must be mapped to a single sensor channel and the other one to two sensor channels,
             e.g. {'x': ['x1', 'x2'], 'y': 'y1'}. The order of the elements in the list with two channels matters for
             the calculation of the rotation angle (in general the left or top sensor should come first).
         distance: (float > 0) distance in m between the sensor heads on the axis with two channels
             (for calculating the rotation).
+        ref_on_init: (True, False, "always", "if necessary", "never")
+            * "always": Run referencing procedure every time the driver is initialized, no matter the state it was in.
+            * True / "if necessary": If the channels are already in a valid state (i.e. the device was not turned off
+                since the last referencing), don't reference again. Only reference if the channels are not valid (i.e.
+                the device was turned off after the last referencing). In any case, the device can be used after
+                the referencing procedure is complete. It is recommended to reference the system frequently though.
+                In case the system has not been power cycled in a long time, the referencing parameters might become
+                outdated and the reported position values might not be accurate.
+            * False / "never": Never reference. This means that the device might not be able to produce position data,
+                if it was not previously referenced.
```

Comment:
I agree with you that strings are typically not a good idea to use. However, this is just too follow other drivers which already use these values. So it's a little historical. Actually, originally it was just True/False, which was fine. However, we had to extend to 3 choices. As you said, the input will come from the YAML file. We cannot rely on constants on the YAML file, so the strings are the next best thing (because they still are understandable by the user).